src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/SlhDsa/SlhDsaTests.cs

```diff
@@ -149,19 +140,17 @@ public void GenerateExportSecretKeySignAndVerify(SlhDsaAlgorithm algorithm)
 
             using (SlhDsa slhDsa = GenerateKey(algorithm))
             {
-                signature = new byte[algorithm.SignatureSizeInBytes];
-                Assert.Equal(signature.Length, slhDsa.SignData(data, signature));
+                signature = slhDsa.SignData(data);
 
-                secretKey = new byte[algorithm.SecretKeySizeInBytes];
-                Assert.Equal(secretKey.Length, slhDsa.ExportSlhDsaSecretKey(secretKey));
+                secretKey = slhDsa.ExportSlhDsaSecretKey();
             }
 
             using (SlhDsa slhDsa = ImportSlhDsaSecretKey(algorithm, secretKey))
             {
                 ExerciseSuccessfulVerify(slhDsa, data, signature, []);
 
                 signature.AsSpan().Clear();
-                Assert.Equal(signature.Length, slhDsa.SignData(data, signature));
+                signature = slhDsa.SignData(data);
```

Comment:
There are a number of places where we call `thing.AsSpan().Clear()`... and immediately on the next line re-assign the local. I think we can get rid of the `Clear` in these cases.

Reply:
Updated to use the span overloads here instead.